test/src/common/plugin/test-plugins/async-resolve-plugin.js

```diff
@@ -1,16 +1,21 @@
 import {BasePlugin} from '../../../../../src/common/plugins';
+import {Utils} from '@playkit-js/playkit-js';
 
 export default class AsyncResolvePlugin extends BasePlugin {
   static DELAY_ASYNC = 500;
   static isValid(): boolean {
     return true;
   }
+  promise: DeferredPromise;
 
+  constructor(name: string, player: Object, config: Object) {
+    super(name, player, config);
+    this.promise = Utils.Object.defer();
```

Comment:
I think it preferred to move it to ready to make sure the timeout will resolve after your check of promise fulfill order happened correctly.

Reply:
Actually i liked this better as it is more like a real life example.
Putting the setTimeout in the getter will make need to add some flagging mechanism and also if I define two and more plugins then the time for them to be ready will the accumulation of their setTimeouts instead of the max setTimeout like in real life (and also how I implemented it now)